packages/office-ui-fabric-react/src/components/MessageBar/examples/MessageBar.Basic.Example.tsx

```diff
@@ -1,160 +1,184 @@
 import * as React from 'react';
-import { MessageBarButton } from 'office-ui-fabric-react/lib/Button';
-import { Link } from 'office-ui-fabric-react/lib/Link';
-import { Label } from 'office-ui-fabric-react/lib/Label';
-import { Stack, StackItem } from 'office-ui-fabric-react/lib/Stack';
-import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
+import { MessageBarButton, Link, Stack, StackItem, MessageBar, MessageBarType, ChoiceGroup } from 'office-ui-fabric-react';
 
-const log = (text: string): (() => void) => (): void => console.log(text);
+export const MessageBarBasicExample: React.StatelessComponent = () => {
+  const [choice, setChoice] = React.useState<string | undefined>(undefined);
+  const resetChoice = () => setChoice(undefined);
 
-export const MessageBarBasicExample: React.StatelessComponent = () => (
-  <Stack tokens={{ childrenGap: 20 }}>
-    <StackItem>
-      <Label>Info/Default MessageBar</Label>
-      <MessageBar>
-        Info lorem ipsum dolor sit amet, a elit sem interdum consectetur adipiscing elit.{' '}
-        <Link href="www.bing.com" target="_blank">
-          Visit our website.
-        </Link>
-      </MessageBar>
-    </StackItem>
+  const DefaultExample = () => (
```

Comment:
These should probably go outside the example component function.

Reply:
Are you planning to address this? In general I don't think we should be demonstrating creating components within components.

Reply:
Most of the MessageBar examples need access to state in order to reset it. It'd be odd to have most of them inside and then a few outside. 

I also don't want to create addition abstractions to pass this resetChoice functionality into the control. 

I don't see this as component inside a component, but rather components inside of a stateful view.

Reply:
ehh....i guess i already have basic abstraction. i can move the callback into props without too much worry.

Still, i'm always concerned with easy to read demos vs best practice code demos